ivy_tests/test_ivy/test_functional/test_core/test_creation.py

```diff
@@ -202,11 +202,32 @@ def _asarray_helper(draw):
             shared_dtype=True,
         )
     )
+    nested_structure = draw(st.booleans())
+    if nested_structure:
+        sh = draw(
+            helpers.lists(
+                x=st.integers(min_value=1, max_value=20), min_size=1, max_size=5
+            )
+        )
+        sh = ivy.Shape(sh)
+        np_array = np.arange(3).astype(x_dtype[0])
```

Comment:
I think we should generate the numpy array using hypothesis not a fixed np array

Reply:
sure